### ISSUE TITLE
isolated declarations: false-positive TS9023 for expando function declared via ambient namespace

### ISSUE THREAD

**Issue by valentinmelusson** [oxc-project/oxc#24975]
### Version

  `oxc-transform@0.127.0`

  ### Reproduction

  ```ts
  import { isolatedDeclarationSync } from 'oxc-transform';

  const source = `
  export function foo(): void {}

  export declare namespace foo {
      let bar: number;
  }

  foo.bar = 42;
  `;

  const result = isolatedDeclarationSync('repro.ts', source, {
      sourcemap: false,
  });

  console.log(result.errors);
  ```

  ### Actual behavior

  OXC reports:

  ```text
  TS9023: Assigning properties to functions without declaring them is not
  supported with --isolatedDeclarations. Add an explicit declaration for the
  properties assigned to this function.
  ```

  ### Expected behavior

  No error should be reported. The `bar` property is explicitly declared through function/namespace declaration merging:

  ```ts
  export declare namespace foo {
      let bar: number;
  }
  ```

  TypeScript 6.0.2 accepts this source with `isolatedDeclarations: true` and emits:

  ```ts
  export declare function foo(): void;

  export declare namespace foo {
      let bar: number;
  }
  ```

  OXC also produces the expected declaration output despite reporting TS9023.

  A non-ambient namespace is accepted by OXC:

  ```ts
  export namespace foo {
      export let bar: number;
  }
  ```

  The error only appears with an ambient namespace:

  ```ts
  export declare namespace foo {
      let bar: number;
  }
  ```

Maybe related to the expando-function support added in [#4910](https://github.com/oxc-project/oxc/pull/4910) ?

**Comment by valentinmelusson** [comment]
👋 Hey @camc314, WDYT of subotac's PR? We are still facing the error 🙏
